backend.md

```diff
@@ -1,75 +1,74 @@
 # Backend Challenge
+> *Este arquivo também está disponível em português [aqui](backend_pt.md).*
 
-## About the Challenge
+In Zé we thrive to find our best partner to deliver beverages to our customers, providing the best and fastest service.
+To achieve this our compute fleet deals with [GIS](https://en.wikipedia.org/wiki/Geographic_information_system) objects all the time.
 
-In this [json](files/pdvs.json) you will find data about hundreds of partners with the following format:
-```javascript
+When programming we try to follow a myriad of coding best practices and patterns (those you can read on books like Clean Code, Clean Architecture, The Pragmatic Programmer, Domain-Driven Design, Microservice Patterns, etc...). 
+Since writing **good code is a must** on our daily basis we expect people that want to join our team think alike. This code challenge is something we designed to find this awesome programmer.
+
+## 1. What we want you to do
+
+We expect you to develop a service that provides an API using REST or GraphQL that can do the three following functionalities and technical requirements:
+
+### 1.1. Create a partner:
+Save in a database a partner defined by **all** the fields represented by the JSON and rules below:
+```json
 {
-  "pdvs": [ 
-    {
-        "id": 1, 
-        "tradingName": "Adega da Cerveja - Pinheiros",
-        "ownerName": "Zé da Silva",
-        "document": "1432132123891/0001", //CNPJ
-        "coverageArea": { 
-          "type": "MultiPolygon", 
-          "coordinates": [
-            [[[30, 20], [45, 40], [10, 40], [30, 20]]], 
-            [[[15, 5], [40, 10], [10, 20], [5, 10], [15, 5]]]
-          ]
-        }, //Área de Cobertura
-        "address": { 
-          "type": "Point",
-          "coordinates": [-46.57421, -21.785741]
-        }, // Localização do PDV
-    }
-  ]
+  "id": 1, 
+  "tradingName": "Adega da Cerveja - Pinheiros",
+  "ownerName": "Zé da Silva",
+  "document": "1432132123891/0001",
+  "coverageArea": { 
+    "type": "MultiPolygon", 
+    "coordinates": [
+      [[[30, 20], [45, 40], [10, 40], [30, 20]]], 
+      [[[15, 5], [40, 10], [10, 20], [5, 10], [15, 5]]]
+    ]
+  },
+  "address": { 
+    "type": "Point",
+    "coordinates": [-46.57421, -21.785741]
+  }
 }
 ```
 
-Below, you can give a look on how these partners might be represented in a map:
+1. The `address` field follows the `GeoJSON Point` format (https://en.wikipedia.org/wiki/GeoJSON);
+2. The `coverageArea` field follows the `GeoJSON MultiPolygon` format (https://en.wikipedia.org/wiki/GeoJSON);
+3. The `document` must be a unique field;
+4. The `id` must be a unique field, but not necessarily an integer;
 
+You can use this [json](files/pdvs.json) file composed of hundreds of partner information to test your application.
```

Comment:
Again, from my own experience with the test:

From the text, as it is written, the `pdvs.json` file is supposed to be used only as a guide. I think we can more emphatic here and go as far as saying "we do not expect these documents to be pre loaded in the database [etc, etc]" or something like that.

Otherwise, if we do expect the documents in the example file to actually be loaded to the database during deploy (or any other process) we have to state it somewhere (perhaps in the "technical requirements" section) that it is a requirement.

Reply:
Also, there are some kinked polygons, and perhaps the candidate will have to do some logic in order to add them to the database.

Reply:
Good point, guys!
Also, we can make a new pdvs.json. 😃